CliWrap/Utils/NativeMethods.cs

```diff
@@ -8,5 +10,258 @@ public static class Unix
+        public static extern int LoginTty(int fd);
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct WinSize
+        {
+            public ushort Row;
+            public ushort Col;
+            public ushort XPixel;
+            public ushort YPixel;
+        }
+
+        // TIOCSWINSZ ioctl request codes (platform-specific)
+        public static nuint TIOCSWINSZ => OperatingSystem.IsMacOS() ? 0x80087467u : 0x5414u;
+
+        // posix_spawn for proper PTY process creation
+        [DllImport("libc", EntryPoint = "posix_spawnp", SetLastError = true)]
+        public static extern int PosixSpawnp(
+            out int pid,
+            string file,
+            IntPtr file_actions,
+            IntPtr attrp,
+            string[] argv,
+            string[] envp
+        );
+
+        [DllImport("libc", EntryPoint = "posix_spawn_file_actions_init", SetLastError = true)]
+        public static extern int PosixSpawnFileActionsInit(IntPtr file_actions);
+
+        [DllImport("libc", EntryPoint = "posix_spawn_file_actions_destroy", SetLastError = true)]
+        public static extern int PosixSpawnFileActionsDestroy(IntPtr file_actions);
+
+        [DllImport("libc", EntryPoint = "posix_spawn_file_actions_adddup2", SetLastError = true)]
+        public static extern int PosixSpawnFileActionsAddDup2(
+            IntPtr file_actions,
+            int fd,
+            int newfd
+        );
+
+        [DllImport("libc", EntryPoint = "posix_spawn_file_actions_addclose", SetLastError = true)]
+        public static extern int PosixSpawnFileActionsAddClose(IntPtr file_actions, int fd);
+
+        [DllImport("libc", EntryPoint = "posix_spawnattr_init", SetLastError = true)]
+        public static extern int PosixSpawnAttrInit(IntPtr attr);
+
+        [DllImport("libc", EntryPoint = "posix_spawnattr_destroy", SetLastError = true)]
+        public static extern int PosixSpawnAttrDestroy(IntPtr attr);
+
+        [DllImport("libc", EntryPoint = "posix_spawnattr_setflags", SetLastError = true)]
+        public static extern int PosixSpawnAttrSetFlags(IntPtr attr, short flags);
+
+        [DllImport("libc", EntryPoint = "waitpid", SetLastError = true)]
+        public static extern int WaitPid(int pid, out int status, int options);
+
+        // posix_spawn flags
+        public const short POSIX_SPAWN_SETSID = 0x80; // Linux-specific, creates new session
+
+        // waitpid options
+        public const int WNOHANG = 1;
+
+        // Size of posix_spawn structures (platform-specific, using max observed size)
+        // Linux: posix_spawn_file_actions_t is 80 bytes, posix_spawnattr_t is 336 bytes
+        // macOS: These are pointers (8 bytes on 64-bit)
+        public static int PosixSpawnFileActionsSize => OperatingSystem.IsMacOS() ? 8 : 80;
+        public static int PosixSpawnAttrSize => OperatingSystem.IsMacOS() ? 8 : 336;
+    }
+
+    public static class Windows
```

Comment:
At this point, let's split each platform into its own partial class (`NativeMethods.Unix.cs`, `NativeMethods.Windows.cs`).